vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/test/java/com/vaadin/flow/component/checkbox/tests/CheckboxGroupTest.java

```diff
@@ -302,6 +302,32 @@ public void allDataRefreshEvent() {
 
     }
 
+    @Test
+    public void selectItem_setItemLabelGenerator_selectionIsRetained() {
+        CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>();
+        checkboxGroup.setItems("foo", "bar");
+
+        checkboxGroup.setValue(Set.of("foo"));
+        Assert.assertEquals(Set.of("foo"), checkboxGroup.getValue());
+
+        checkboxGroup.setItemLabelGenerator(item -> item + " (Updated)");
+
+        Assert.assertEquals(Set.of("foo"), checkboxGroup.getValue());
+    }
+
+    @Test
+    public void selectItem_setItemLabelGenerator_labelIsUpdated() {
```

Comment:
```suggestion
    public void setItemLabelGenerator_labelIsUpdated() {
```
There isn't any selection happening in this test case.